### ISSUE TITLE
Streaming select Flood Gage Data with Alarm Setting based on Normal Level & Rate of Change

### ISSUE THREAD

**Issue by judopp** [sketch-city/project-ideas#84]
People who have lived through one or more floods/hurricanes realize that data and evacuation lead time are essential to getting through an emergency unscathed.  One is better off staying put provided that flooding does not occur.  While such data is out there, it requires scrolling through several pages and looking at a map of hundreds of instruments; then picking the one you are interested in.  This map provides either a snapshot of levels around the Houston Metro area or, if you plug in other time criteria, a time-weighted average.  

What people in flood-prone areas need is real-time data plotted alongside normal readings.  When levels are increasing at rates that suggest flooding is imminent, some kind of "Amber-Alert" type message should go out to the user.

**Comment by airpaio** [comment]
Hello @judopp 
I would like to be involved in this. 

I actually found this sketch-city repo for the first time today, and there are some nice ideas here. Luckily my home was spared by the 2016 Tax Day floods by just a couple hundred feet in the Spring area. Others in my neighborhood weren't so lucky. This issue has been on my mind since then.

For the "Amber-Alert" type of notification, I would like to add that this type of notification could have been a life saver for drivers in the tragic events that they drove into flooded underpasses or creeks because they lost track of the flooded road they were on. 

There was a similar idea going on in issue #28 but I dont know how far they got with the idea.

I'll start looking over the data sources to see how we might get and organize it.

**Comment by judopp** [comment]
Cory,
I am going to set up a Channel on the Slack-site so that we can communicate more seamlessly.


The plan is to put together a demo based on a "Minimum Viable Product".  Here, we would give the user a menu of 10 flood gages to chose from.  For this version, only one gage could be viewed at a time.  We would need to obtain the historical data and be able to plot it including calculated values for Min, Max (over a much larger time period) and Average (for normal or steady state operation).  


Do you know how to obtain streaming data from the internet?  I am taking a course and won't be skilled in this for another few weeks.  Will you be at the next Tuesday night meeting?  We need to meet and plan our work going forward.  


Thanks for reaching out,
Judith



-----Original Message-----

**Comment by judopp** [comment]
Cory,


Please sign up for Slack at this link:  sketchcity.herokuapp.com.


This site will be a way for you to more easily communicate and share your findings with me and others who join the team.  
Thanks,
Judith





-----Original Message-----

**Comment by airpaio** [comment]
@judopp I have pulled streaming data before. I'll have to dig up the code and should be able to fix it to pull the flood guage data. 

I wont be able to make it to the Tuesday meeting; I have some other commitments to take care of. I can try to schedule another time for a face to face meeting or I'm even more flexible with scheduling a video chat. I'll sign on to slack a bit later today and we'll go from there.
-Cory

**Comment by judopp** [comment]
Cory,
I understand.  We can definitely meet somewhere - Face-to-face is good for the first get together.  After that, we can do the video chat.  Please schedule this - keep in mind that I live close to NASA so would prefer to avoid morning and evening rush-hours.  Also, I will be out of town from 3/29 through 4/2.  


Until then,
Judith



-----Original Message-----

**Comment by judopp** [comment]
Here is the link that I found:  https://www.harriscountyfws.org

**Comment by airpaio** [comment]
Thats the one ive been looking at. I've been studying how the webpage is put together so I can scrape data from it. Too bad we dont have direct access to their database :(

We can get two types of data, rainfall and stream elevation. 

And it seems we can choose historical data down to the minute, and I'm not sure how far back in time. I think it would be dependent of which station, and when that station was installed.

I'm wondering what kind of time interval we would want? In a heavy rain, I think a stream could spill over its banks in just a few minutes time, so maybe a 5 minute interval would be sufficient?

By the way, I joined your slack channel, so we can chat on there now.

**Comment by judopp** [comment]
I posted this on #84 and responded.  Also invited Jeff and provided him with the gage map.  



-----Original Message-----

**Comment by fileunderjeff** [comment]
People interested in this project should join the flood gauge channel on the Sketch City slack: http://sketchcity.herokuapp.com

**Comment by paragmchaudhari** [comment]
Is this thread abandoned?

**Comment by judopp** [comment]
Yes.  I read that there is an app that does this.  Linda Hidalgo mentioned it at one of her meetings.
